CMakeLists.txt

```diff
@@ -275,10 +275,17 @@ endif()
 if(MINGW)
   set_property(TARGET SDL2 APPEND_STRING PROPERTY LINK_FLAGS " -nostdlib")
   if(USE_CLANG)
-    set_property(TARGET SDL2 APPEND_STRING PROPERTY LINK_FLAGS " -rtlib=libgcc")
+    if(SDL_CPU_X86)
+      target_link_libraries(SDL2 PRIVATE clang_rt.builtins-i386)
+    elseif(SDL_CPU_x64)
+      target_link_libraries(SDL2 PRIVATE clang_rt.builtins-x86_64)
+    else()
+      message(WARNING "Unknown clang_rt library (arch=${SDL_CPU_NAMES})")
+    endif()
+  else()
+    target_link_libraries(SDL2 PRIVATE "gcc")
+    set_property(TARGET SDL2 APPEND_STRING PROPERTY LINK_FLAGS " -static-libgcc") # libgcc is needed for 32 bit (x86) builds
```

Comment:
No, we should not link in static libgcc, we do not include the LGPL nor do we require that applications that use our code comply with the LGPL.

Reply:
SDL3 does this. So it should be removed there too.

Reply:
Yes, definitely.

Reply:
Does using clang introduce any runtime dependencies or licensing requirements?

Reply:
Yes

Reply:
What are they?

Reply:
> Yes, definitely.

Then how will you resolve compiler runtime symbols?

> not include the LGPL nor do we require that applications that use our code comply with the LGPL.

IIRC, libgcc is LGPL3 but with "additional permissions described in the GCC Runtime Library Exception"
https://gcc.gnu.org/git/?p=gcc.git;a=blob;f=COPYING.RUNTIME;hb=HEAD

Reply:
> SDL3 does this. So it should be removed there too.

SDL2 does that too, _for years_

Reply:
Ahhh, the runtime library exception is what we need here. What does it cover?

Reply:
> Ahhh, the runtime library exception is what we need here. What does it cover?

Not a lawyer.. You should read that doc yourself.

Reply:
[compiler-rt is Apache 2.0 with LLVM Exception] (https://packages.msys2.org/package/mingw-w64-clang-i686-compiler-rt?repo=clang32).

Reply:
Okay, reading through it looks like both libgcc and compiler-rt are covered by exceptions that allow us to link them. Thank you for the clarification, we should be good to use them statically here.